src/main/java/fi/metatavu/metamind/bot/functions/KuntaApiGetEventsMetaBotFunction.java

```diff
@@ -0,0 +1,190 @@
+import com.rabidgremlin.mutters.core.IntentMatch;
+import com.rabidgremlin.mutters.core.session.Session;
+
+import fi.metatavu.metamind.kuntaapi.Events;
+
+/**
+ * Metabot function to get events from KuntaAPI
+ * 
+ * @author Ville Koivukangas
+ */
+@ApplicationScoped
+public class KuntaApiGetEventsMetaBotFunction extends AbstractMetaBotFunction {
+  @Inject
+  private Logger logger;
+
+  @Override
+  public String getFunctionName() {
+    return "KUNTA_API_GET_EVENTS";
+  }
+
+  @Override
+  public void execute(CurrentResponse currentResponse, Session session, IntentMatch intentMatch, Story story,
+      String param) {
+    FunctionDetails details = FunctionHelper.parseFunctionString(param);
+
+    String authorizationToken = (String) session.getLongTermAttribute("authorizationToken");
+    String organizationId = details.getFunctionParams().get("organizationId");
+    String route = "events";
+    String routeId = "";
+
+    ArrayList<Events> events = new ArrayList<Events>();
+
+    StringBuffer content = null;
+    HttpURLConnection connection = null;
+
+    if (!organizationId.isEmpty() || organizationId != null) {
+      connection = createHttpConnection(authorizationToken, organizationId, route, routeId);
+      content = getContent(connection);
+      events = parseContent(content);
+      connection.disconnect();
+    }
+
+    Random random = new Random();
+    int rand = random.nextInt(events.size());
+
+    try {
+      story.getVariablesState().set("eventName", events.get(rand).getName());
+      story.getVariablesState().set("eventDescription", events.get(rand).getDescription());
+      story.getVariablesState().set("eventLocation", events.get(rand).getPlace());
+      story.getVariablesState().set("eventAddress", events.get(rand).getAddress());
+      story.getVariablesState().set("eventZip", events.get(rand).getZip());
+      story.getVariablesState().set("eventCity", events.get(rand).getCity());
+    } catch (Exception e) {
+      logger.error("Error while setting variables at KUNTA_API_GET_EVENTS", e);
+    }
+  }
+
+  /**
+   * This method returns a HttpURLConnection based on organizationId, route and
+   * routeId.
+   * 
+   * @param organizationId Organization id of kunta-api oranization.
+   * @param route String The route where to get data.
+   * @param routeId (Id or blank String) Route id to get specific data by id.
+   * @return A HttpURLConnection.
+   */
+  private HttpURLConnection createHttpConnection(String authorizationToken, String organizationId, String route,
+      String routeId) {
+    URL url = null;
+
+    try {
+      url = new URL(String.format("http://api.kunta-api.fi/v1/organizations/%s/%s/%s", organizationId, route, routeId));
+    } catch (MalformedURLException e) {
+      logger.error("Failed to create URL at KUNTA_API_GET_EVENTS");
+    }
+
+    HttpURLConnection connection = null;
+
+    try {
+      connection = (HttpURLConnection) url.openConnection();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "url" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)